modules/launcher/src/main/java/org/mule/module/launcher/log4j2/ArtifactAwareContextSelector.java

```diff
@@ -180,10 +183,30 @@ public void execute()
 
     private URI getArtifactLoggingConfig(ArtifactClassLoader muleCL)
```

Comment:
extract the entire buildContext() method (which includes this sub rutine) into a factory which you can unit test using mock application descriptors and classloaders. In that way you can test all this logic without relying on classpath discovery of test log4j config files which we know ends up in flackyness